tests/multiple_database/tests.py

```diff
@@ -1302,6 +1302,31 @@ def test_related_manager(self):
             title="Dive into Water", published=datetime.date(2009, 5, 4), extra_arg=True
         )
 
+    @override_settings(DATABASE_ROUTERS=["multiple_database.tests.TestRouter"])
+    def test_contenttype_in_separate_db(self):
+        ContentType.objects.using("other").all().delete()
+        book_title = "Test Content Type in Separate DB"
+        book_published_date = datetime.date(2009, 5, 4)
+        Book.objects.using("other").create(
+            title=book_title, published=book_published_date
+        )
+
+        book_type = ContentType.objects.using("default").get(
+            app_label="multiple_database", model="book"
+        )
+
+        book = book_type.get_object_for_this_type(title=book_title)
+        self.assertEqual(book.published, book_published_date)
+
+        book = book_type.get_object_for_this_type(using="other", title=book_title)
+        self.assertEqual(book.published, book_published_date)
+
+        with self.assertRaises(Book.DoesNotExist):
+            book_type.get_object_for_this_type(using="default", title=book_title)
+
+        all_books = book_type.get_all_objects_for_this_type()
+        self.assertEqual(len(all_books), 1)
```

Comment:
I'd add a second book in the `default` db to confirm how it behaves when books are in two different databases. I'd expect that something like this to work:
```suggestion
        ContentType.objects.using("other").all().delete()
        
        book_other = Book.objects.using("other").create(
            title="Test title other", published=datetime.date(2009, 5, 4)
        )
        book_default = Book.objects.using("other").create(
            title="Test title default", published=datetime.date(2009, 5, 4)
        )
        book_type = ContentType.objects.using("default").get(
            app_label="multiple_database", model="book"
        )

        book = book_type.get_object_for_this_type(title=book_other.title)
        self.assertEqual(book, book_other)
        book = book_type.get_object_for_this_type(using="other", title=book_other.title)
        self.assertEqual(book, book_other)
        
        book = book_type.get_object_for_this_type(title=book_default.title)
        self.assertEqual(book, book_default)
        book = book_type.get_object_for_this_type(using="default", title=book_default.title)
        self.assertEqual(book, book_default)

        all_books = book_type.get_all_objects_for_this_type()
        self.assertCountEqual(all_books, [book_default, book_other])
```

Reply:
Moreover, ticket-25564 and ticket-30760 were marked as a duplicates of this ticket. Can you add a regression test for them to confirm that the "View on site" is also fixed?

Reply:
It looks like ticket-15610 may be another duplicate, Can you confirm this and add an appropriate regression test?

Reply:
@felixxm I tried your additional tests, but they don't work currently. `book_type.get_object_for_this_type(title=book_default.title)` isn't able to find the book in the default database, and `book_type.get_all_objects_for_this_type()` doesn't check through both databases to find all the objects.

Is it expected that these queries will check across all databases to find objects? Seems like we could add code to fall back to other databases for `get_object_for_this_type()`, and we could query all databases in `get_all_objects_for_this_type()`, but would we run into issues with duplicate primary keys if we're adding together lists of objects from multiple databases?

Reply:
@felixxm I added a test for View on Site.

For ticket 15610, I added a comment with a question on that issue.

Reply:
> @felixxm I tried your additional tests, but they don't work currently. `book_type.get_object_for_this_type(title=book_default.title)` isn't able to find the book in the default database, and `book_type.get_all_objects_for_this_type()` doesn't check through both databases to find all the objects.
> 
> Is it expected that these queries will check across all databases to find objects? Seems like we could add code to fall back to other databases for `get_object_for_this_type()`, and we could query all databases in `get_all_objects_for_this_type()`, but would we run into issues with duplicate primary keys if we're adding together lists of objects from multiple databases?

Agreed we cannot check all databases, so it should work as follows:
```python
    @override_settings(DATABASE_ROUTERS=["multiple_database.tests.TestRouter"])
    def test_contenttype_in_separate_db(self):
        book_other = Book.objects.using("other").create(
            title="Test title other", published=datetime.date(2009, 5, 4)
        )
        book_default = Book.objects.using("default").create(
            title="Test title default", published=datetime.date(2009, 5, 4)
        )
        book_type = ContentType.objects.using("default").get(
            app_label="multiple_database", model="book"
        )
        book = book_type.get_object_for_this_type(title=book_other.title)
        self.assertEqual(book, book_other)
        book = book_type.get_object_for_this_type(using="other", title=book_other.title)
        self.assertEqual(book, book_other)
        
        with self.assertRaises(Book.DoesNotExist):
            book = book_type.get_object_for_this_type(title=book_default.title)
        book = book_type.get_object_for_this_type(using="default", title=book_default.title)
        self.assertEqual(book, book_default)
        
        all_books = book_type.get_all_objects_for_this_type()
        self.assertCountEqual(all_books, [book_other])

```

Reply:
OK, I updated that test - hopefully it looks OK now.